torch/_inductor/ir.py

```diff
@@ -2594,6 +2595,55 @@ def codegen_reference(self, writer=None):
         )
 
 
+@dataclasses.dataclass
+class DtypeView(BaseView):
+    """Pretend our storage has a different type"""
+
+    target_dtype: torch.dtype
+
+    def __post_init__(self):
+        super().__post_init__()
+        if isinstance(self.data, BaseView):
```

Comment:
This is copy-pasta'd from ReinterpretView, but the reason ReinterpretView is able to do this is because we know that the input is representable as Storage and Layout